### DISCUSSION TITLE
pwa Next.js

### DISCUSSION THREAD

**Discussion by Wilfried02** [vercel/next.js#48523] [Help] [answered]
### Summary

Next js app can correctly work as progressive web app ??

**Comment by icyJoseph** [comment]
Yes.

However, there's one catch.

Most PWA tools use a strategy called, stale-while-revalidate, https://developer.chrome.com/docs/workbox/modules/workbox-strategies/#stale-while-revalidate - Please do read and understand its implications.

Since stale-while-revalidate, favours showing already existing data, it defeats the purpose of `GetServerSideProps` which wants to show the latest data on demand!

So for pages that use `GetServerSideProps`, we ought to use https://developer.chrome.com/docs/workbox/modules/workbox-strategies/#network-first-network-falling-back-to-cache - again, please do read and understand the implications.

I answered a few times on how to do this for `next-pwa`: https://github.com/vercel/next.js/discussions/42405#discussioncomment-4046461

Last but not least: https://twitter.com/sanketsahu/status/1133223194888773632

**Reply by Wilfried02** [reply]
thank you very much

**Reply by icyJoseph** [reply]
Please don't hesitate to ask more, and if you feel good about the answer, please mark it as answered :)

**Reply by Wilfried02** [reply]
thanks

**Comment by leerob** [comment, marked as answer]
Hey everyone. We now have updated documentation on building PWAs with Next.js.

https://nextjs.org/docs/app/building-your-application/configuring/progressive-web-apps
